src/modules/cpu/host_fused_functions.hpp

```diff
@@ -4,12 +4,593 @@
+                              -.587, .413, -.588, 0.0,
+                              -.114, -.114, .886, 0.0,
+                              0.0, 0.0, 0.0, 0.0};
+
+        Rpp32f ssh_mat[16] = {.168, -.328, 1.25, 0.0,
+                              .330, .035, -1.05, 0.0,
+                              -.497, .292, -.203, 0.0,
+                              0.0, 0.0, 0.0, 0.0};
+
+        Rpp32f sch = saturation * cos(hue * PI_OVER_180);
+        Rpp32f ssh = saturation * sin(hue * PI_OVER_180);
+
+        __m128 psch = _mm_set1_ps(sch);
+        __m128 pssh = _mm_set1_ps(ssh);
+        __m128 p0, p1, p2, p3;
+
+        for (int i = 0; i < 16; i+=4)
+        {
+            p0 = _mm_loadu_ps(hue_saturation_matrix + i);
+            p1 = _mm_loadu_ps(sch_mat + i);
+            p2 = _mm_loadu_ps(ssh_mat + i);
+
+            p3 = _mm_add_ps(_mm_mul_ps(psch, p1), _mm_mul_ps(pssh, p2));
+            p3 = _mm_add_ps(p0, p3);
+
+            _mm_storeu_ps(hue_saturation_matrix + i, p3);
+        }
+
+        Rpp32f scale = contrast + 1.0;
+        Rpp32f brightness_contrast_matrix[16] = {scale, 0., 0., 0.,
+                                                 0., scale, 0., 0.,
+                                                 0., 0., scale, 0.,
+                                                 brightness, brightness, brightness, 1.};
+
+        Rpp32f ctm[16] = {1., 0., 0., 0.,
+                          0., 1., 0., 0.,
+                          0., 0., 1., 0.,
+                          0., 0., 0., 1.};
+
+        fast_matmul4x4_sse(hue_saturation_matrix, brightness_contrast_matrix, ctm);
+
+        T *srcPtrTemp, *dstPtrTemp;
+        srcPtrTemp = srcPtr;
+        dstPtrTemp = dstPtr;
+
+        Rpp32u increment = (srcSizeMax.width - srcSize.width) * channel;
+
+
+
+
+
+        // Method 0 - WORKING - No SSE (compiler optimized) - 0.000789s for average of 100 runs on 8 224x224 images
+
+        Rpp32u bufferLength = srcSize.width * channel;
+
+        for (int i = 0; i < srcSize.height; i++)
+        {
+            int vectorLoopCount = 0;
+            for (; vectorLoopCount < bufferLength; vectorLoopCount+=3)
+            {
+                Rpp32f pixelR = ctm[0] * srcPtrTemp[0] + ctm[1] * srcPtrTemp[1] + ctm[2] * srcPtrTemp[2] + ctm[3];
+                Rpp32f pixelG = ctm[4] * srcPtrTemp[0] + ctm[5] * srcPtrTemp[1] + ctm[6] * srcPtrTemp[2] + ctm[7];
+                Rpp32f pixelB = ctm[8] * srcPtrTemp[0] + ctm[9] * srcPtrTemp[1] + ctm[10] * srcPtrTemp[2] + ctm[11];
+
+                *dstPtrTemp = (T) RPPPIXELCHECK(pixelR);
+                *(dstPtrTemp + 1) = (T) RPPPIXELCHECK(pixelG);
+                *(dstPtrTemp + 2) = (T) RPPPIXELCHECK(pixelB);
+
+                srcPtrTemp += 3;
+                dstPtrTemp += 3;
+            }
+            srcPtrTemp += increment;
+            dstPtrTemp += increment;
+        }
+
+
+
+
+
+        // Method 1 - WORKING - 12 elements per loop run with SSE - 0.001457s for average of 100 runs on 8 224x224 images
```

Comment:
remove all the variations except the most efficient one for reference

Reply:
Done. Just Method 0 will be used for now.